test/test_visualization.jl

```diff
@@ -0,0 +1,76 @@
+module TestVisualization
+
+using Test
+using Documenter
+using Trixi
+using RecipesBase
+
+# Start with a clean environment: remove Trixi output directory if it exists
+outdir = "out"
+isdir(outdir) && rm(outdir, recursive=true)
+
+# Run various visualization tests
+@testset "Visualization tests" begin
+  # Run Trixi
+  @test_nowarn trixi_include(@__MODULE__, joinpath(examples_dir(), "2d", "elixir_euler_blast_wave_amr.jl"),
+                             tspan=(0,0.1))
+
+  @testset "PlotData2D, PlotDataSeries2D, PlotMesh2D" begin
+    # Constructor
+    @test PlotData2D(sol) isa PlotData2D
+    pd = PlotData2D(sol)
+
+    # show
+    @test_nowarn show(stdout, pd)
+    println(stdout)
+
+    # getindex
+    @test pd["rho"] == Trixi.PlotDataSeries2D(pd, 1)
+    @test pd["v1"] == Trixi.PlotDataSeries2D(pd, 2)
+    @test pd["v2"] == Trixi.PlotDataSeries2D(pd, 3)
+    @test pd["p"] == Trixi.PlotDataSeries2D(pd, 4)
+    @test_throws KeyError pd["does not exist"]
+
+    # convenience methods for mimicking a dictionary
+    @test pd[begin] == Trixi.PlotDataSeries2D(pd, 1)
+    @test pd[end] == Trixi.PlotDataSeries2D(pd, 4)
+    @test length(pd) == 4
+    @test size(pd) == (4,)
+    @test keys(pd) == ("rho", "v1", "v2", "p")
+    @test eltype(pd) == Pair{String, Trixi.PlotDataSeries2D}
+    @test [v for v in pd] == ["rho" => Trixi.PlotDataSeries2D(pd, 1),
+                              "v1" => Trixi.PlotDataSeries2D(pd, 2),
+                              "v2" => Trixi.PlotDataSeries2D(pd, 3),
+                              "p" => Trixi.PlotDataSeries2D(pd, 4)]
+
+    # PlotDataSeries2D
+    pds = pd["p"]
+    @test pds.plot_data == pd
+    @test pds.variable_id == 4
+    @test_nowarn show(stdout, pds)
+    println(stdout)
+
+    # getmesh/PlotMesh2D
+    @test getmesh(pd) == Trixi.PlotMesh2D(pd)
+    @test getmesh(pd).plot_data == pd
+    @test_nowarn show(stdout, getmesh(pd))
+    println(stdout)
+  end
+
+  @testset "plot recipes" begin
+    pd = PlotData2D(sol)
+
+    @test RecipesBase.apply_recipe(Dict{Symbol,Any}(), sol) isa Vector{RecipesBase.RecipeData}
+    @test RecipesBase.apply_recipe(Dict{Symbol,Any}(), pd) isa Vector{RecipesBase.RecipeData}
+    @test RecipesBase.apply_recipe(Dict{Symbol,Any}(), pd["p"]) isa Vector{RecipesBase.RecipeData}
+    @test RecipesBase.apply_recipe(Dict{Symbol,Any}(), getmesh(pd)) isa Vector{RecipesBase.RecipeData}
+  end
```

Comment:
This seems to test some lower-level implementation details of RecipesBase.jl. I'm not sure whether it really help us here. Instead, I would like to see some tests using Plots.jl directly. For example, 
```bash
julia --project=test test/test_visualization.jl
```
passes but I get
```julia
julia> using Trixi, Plots

julia> trixi_include(joinpath(examples_dir(), "2d", "elixir_hypdiff_lax_friedrichs.jl"))
[...]

julia> plot(sol)
ERROR: Cannot convert NamedTuple{(:t, :u, :prob),Tuple{Tuple{Float64,Float64},Tuple{Array{Float64,1},Array{Float64,1}},ODEProblem{Array{Float64,1},Tuple{Float64,Float64},true,SemidiscretizationHyperbolic{TreeMesh{2,Trixi.SerialTree{2}},HyperbolicDiffusionEquations2D{Float64},typeof(initial_condition_poisson_periodic),Trixi.BoundaryConditionPeriodic,typeof(source_terms_poisson_periodic),DG{Float64,LobattoLegendreBasis{Float64,4,Array{Float64,2},StaticArrays.SArray{Tuple{4,2},Float64,2,8},StaticArrays.SArray{Tuple{4,4},Float64,2,16}},Trixi.LobattoLegendreMortarL2{Float64,4,StaticArrays.SArray{Tuple{4,4},Float64,2,16}},typeof(flux_lax_friedrichs),VolumeIntegralWeakForm},NamedTuple{(:elements, :interfaces, :boundaries, :mortars, :fstar_upper_threaded, :fstar_lower_threaded),Tuple{Trixi.ElementContainer2D{Float64,3,3},Trixi.InterfaceContainer2D{Float64,3,3},Trixi.BoundaryContainer2D{Float64,3,3},Trixi.L2MortarContainer2D{Float64,3,3},Array{StaticArrays.MArray{Tuple{3,4},Float64,2,12},1},Array{StaticArrays.MArray{Tuple{3,4},Float64,2,12},1}}}},ODEFunction{true,typeof(Trixi.rhs!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},DiffEqBase.StandardODEProblem}}} to series data for plotting
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] _prepare_series_data(::NamedTuple{(:t, :u, :prob),Tuple{Tuple{Float64,Float64},Tuple{Array{Float64,1},Array{Float64,1}},ODEProblem{Array{Float64,1},Tuple{Float64,Float64},true,SemidiscretizationHyperbolic{TreeMesh{2,Trixi.SerialTree{2}},HyperbolicDiffusionEquations2D{Float64},typeof(initial_condition_poisson_periodic),Trixi.BoundaryConditionPeriodic,typeof(source_terms_poisson_periodic),DG{Float64,LobattoLegendreBasis{Float64,4,Array{Float64,2},StaticArrays.SArray{Tuple{4,2},Float64,2,8},StaticArrays.SArray{Tuple{4,4},Float64,2,16}},Trixi.LobattoLegendreMortarL2{Float64,4,StaticArrays.SArray{Tuple{4,4},Float64,2,16}},typeof(flux_lax_friedrichs),VolumeIntegralWeakForm},NamedTuple{(:elements, :interfaces, :boundaries, :mortars, :fstar_upper_threaded, :fstar_lower_threaded),Tuple{Trixi.ElementContainer2D{Float64,3,3},Trixi.InterfaceContainer2D{Float64,3,3},Trixi.BoundaryContainer2D{Float64,3,3},Trixi.L2MortarContainer2D{Float64,3,3},Array{StaticArrays.MArray{Tuple{3,4},Float64,2,12},1},Array{StaticArrays.MArray{Tuple{3,4},Float64,2,12},1}}}},ODEFunction{true,typeof(Trixi.rhs!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},DiffEqBase.StandardODEProblem}}}) at ~/.julia/packages/RecipesPipeline/uPBKQ/src/series.jl:8
 [3] _series_data_vector(::NamedTuple{(:t, :u, :prob),Tuple{Tuple{Float64,Float64},Tuple{Array{Float64,1},Array{Float64,1}},ODEProblem{Array{Float64,1},Tuple{Float64,Float64},true,SemidiscretizationHyperbolic{TreeMesh{2,Trixi.SerialTree{2}},HyperbolicDiffusionEquations2D{Float64},typeof(initial_condition_poisson_periodic),Trixi.BoundaryConditionPeriodic,typeof(source_terms_poisson_periodic),DG{Float64,LobattoLegendreBasis{Float64,4,Array{Float64,2},StaticArrays.SArray{Tuple{4,2},Float64,2,8},StaticArrays.SArray{Tuple{4,4},Float64,2,16}},Trixi.LobattoLegendreMortarL2{Float64,4,StaticArrays.SArray{Tuple{4,4},Float64,2,16}},typeof(flux_lax_friedrichs),VolumeIntegralWeakForm},NamedTuple{(:elements, :interfaces, :boundaries, :mortars, :fstar_upper_threaded, :fstar_lower_threaded),Tuple{Trixi.ElementContainer2D{Float64,3,3},Trixi.InterfaceContainer2D{Float64,3,3},Trixi.BoundaryContainer2D{Float64,3,3},Trixi.L2MortarContainer2D{Float64,3,3},Array{StaticArrays.MArray{Tuple{3,4},Float64,2,12},1},Array{StaticArrays.MArray{Tuple{3,4},Float64,2,12},1}}}},ODEFunction{true,typeof(Trixi.rhs!),LinearAlgebra.UniformScaling{Bool},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},DiffEqBase.StandardODEProblem}}}, ::Dict{Symbol,Any}) at ~/.julia/packages/RecipesPipeline/uPBKQ/src/series.jl:27
 [4] macro expansion at ~/.julia/packages/RecipesPipeline/uPBKQ/src/series.jl:145 [inlined]
 [5] apply_recipe(::AbstractDict{Symbol,Any}, ::Type{RecipesPipeline.SliceIt}, ::Any, ::Any, ::Any) at ~/.julia/packages/RecipesBase/92zOw/src/RecipesBase.jl:282
 [6] _process_userrecipes!(::Any, ::Any, ::Any) at ~/.julia/packages/RecipesPipeline/uPBKQ/src/user_recipe.jl:36
 [7] recipe_pipeline!(::Any, ::Any, ::Any) at ~/.julia/packages/RecipesPipeline/uPBKQ/src/RecipesPipeline.jl:70
 [8] _plot!(::Plots.Plot, ::Any, ::Any) at ~/.julia/packages/Plots/vsE7b/src/plot.jl:172
 [9] #plot#129 at ~/.julia/packages/Plots/vsE7b/src/plot.jl:58 [inlined]
 [10] plot(::Any) at ~/.julia/packages/Plots/vsE7b/src/plot.jl:52
 [11] top-level scope at REPL[4]:1
```

Reply:
Okay, I figured out what's going on here. This is caused by our own time integrator since we use
```julia
sol = Trixi.solve(ode, Trixi.HypDiffN3Erk3Sstar52(), ...)
```
in this recipe. Thus, we have some options here
- Our ad-hoc time integrators will not support `plot(sol)` :cry: 
- We create our own solution type mimicking `ODESolution` and make it work with `plot(sol)`
- We return an `ODESolution` from our own integrators.

Reply:
Possible fix in e6a66445.